backend/fms_core/services/library.py

```diff
@@ -48,3 +51,96 @@ def create_library(library_type, index, platform, strandedness, library_size=Non
+        errors.append(f"Execution date is not valid.")
+
+    # Retrieve library object linked to the source sample
+    library_source_obj = sample_source.derived_sample_not_pool.library
+    if not sample_source.is_library or library_source_obj is None:
+        errors.append(f"Sample in container {sample_source.name} is not a library.")
+    elif library_source_obj.platform == platform:
+        errors.append(f"Source library platform and destination library platform can't be the same.")
+
+    if not errors:
+        # Create new destination library
+        library_destination, errors_library_destination, warnings_library_destinations = \
+            create_library(library_type=library_source_obj.library_type,
+                           # TODO: Verify this with Janick
+                           library_size=library_size,
+                           index=library_source_obj.index,
+                           platform=platform,
+                           strandedness=SINGLE_STRANDED)
+        try:
+            sample_source.volume = sample_source.volume - volume_used
+            sample_source.save()
+
+            library_destination_data = dict(
+                container_id=container_destination.id,
+                coordinates=coordinates_destination if coordinates_destination else "",
+                creation_date=execution_date,
+                # TODO: Verify this with Janick
+                concentration=concentration if concentration is not None else None,
+                volume=volume_destination if volume_destination is not None else volume_used,
+                depleted=False,
+                # Reset QC flags
+                quantity_flag=None,
+                quality_flag=None
+            )
+
+            new_derived_sample_data = {
+                "library_id": library_destination.id
+            }
+            derived_samples_destination = []
+            volume_ratios = {}
+            for derived_sample_source in sample_source.derived_samples.all():
+                new_derived_sample, errors_inherit, warnings_inherit = inherit_derived_sample(derived_sample_source,
+                                                                                              new_derived_sample_data)
+                errors.extend(errors_inherit)
+                warnings.extend(warnings_inherit)
+
+                derived_samples_destination.append(new_derived_sample)
+                volume_ratios[new_derived_sample.id] = DerivedBySample.objects.get(sample=sample_source,
+                                                                                   derived_sample=derived_sample_source).volume_ratio
+
+            library_destination, errors_process, warnings_process = _process_sample(process,
+                                                                                   sample_source,
+                                                                                   library_destination_data,
```

Comment:
name is confusing ... might be better called sample_destination_data since we have a library object.